### PR TITLE
feat: initial version

### PR DESCRIPTION
**Pull request by mhdawson**
## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by mhdawson** [comment]
What this needs before it can work/land is

1. Publishes of the ubi-base-stack which should occur one this PR lands - https://github.com/paketo-community/ubi-base-stack/pull/3
1. An update to https://github.com/paketo-community/ubi-nodejs-extension to use the images published in 1)
1. Releases for https://github.com/paketo-community/ubi-nodejs-extension

File: .github/CODEOWNERS
Comment (ForestEckhardt) [inline .github/CODEOWNERS:1]:
```suggestion
* @paketo-community/builders-maintainers
```

File: builder.toml
Comment (mhdawson) [inline builder.toml:27]:
```suggestion
  build-image = "docker.io/paketocommunity/build-ubi-base:0.0.1"
```

**Comment by mhdawson** [comment]
Everything needed for this to land is now done -  @paketo-community/builders-maintainers.

I validated that with the latest commit I've added the smoke tests run/pass.

File: builder.toml
Comment (mhdawson) [inline builder.toml:29]:
```suggestion
  build-image = "docker.io/paketocommunity/build-ubi-base:0.0.1"
  run-image = "index.docker.io/paketocommunity/run-ubi-base:latest"
  run-image-mirrors = []
```

File: go.mod
Comment (robdimsdale) [inline go.mod:3]: We should probably start off with `go 1.20` given that `go 1.18` is likely to go out of support shortly.

File: smoke/go_test.go
Comment (robdimsdale) [inline smoke/go_test.go:1]: Are we actually using this test? It seems like `init_test.go` only invokes the nodejs test.

**Review by robdimsdale — COMMENTED** [review]
Mostly looks good, thanks. A couple of questions inline.

**Comment by ForestEckhardt** [comment]
Additionally I am having trouble running the smoke tests because extensions are still an experimental feature in `pack v0.29.0` is there any work around to get the tests to run so that we can validate that things are functioning properly?

**Comment by ForestEckhardt** [comment]
@mhdawson I was able to get experimental mode working during build but I am seeing a panic during `pack builder create` so I am not able to get a successful run of the smoke tests yet

**Comment by mhdawson** [comment]
@ForestEckhardt I'll re-run the smoke tests locally to see what is going on and also update the scripts to set the required command to enable experimental options.

File: go.mod
Comment (mhdawson) [reply go.mod:3]: will update and test.

File: smoke/go_test.go
Comment (mhdawson) [reply smoke/go_test.go:1]: You are correctly I removed all of the tests other than the Node.js one since the builder initially will only have Node.js support.

I'm a bit surprised go did not complain about it, since it complains about most un-used things. Will remove in next commit

**Comment by mhdawson** [comment]
@ForestEckhardt figured out that I had a later version of pack in by path and that was overriding what was in scripts/.util/tools.json. Updated that and it should work now. 

I did also discover that it seems to work with version v0.30.0-pre2 but not v0.30.0-pre3. Will open issue in pack repo to discuss.

Also should have addressed all comments from @robdimsdale as well.

**Comment by mhdawson** [comment]
Issue in pack repo for problems with v0.30.0-pre3 - https://github.com/buildpacks/pack/issues/1850

File: go.mod
Comment (mhdawson) [reply go.mod]: done

**Comment by ForestEckhardt** [comment]
Alrighty I have gotten the smoke tests to pass. @robdimsdale I would like to get you approval on this as well and then we can merge away.

**Review by ForestEckhardt — APPROVED** [review]

**Review by robdimsdale — APPROVED** [review]
This generally looks good. I realized that we'll need to add the `tools.json` to the `.syncignore` file otherwise the pack version will be automatically reset to the latest published version. That's inline.

Let's make that change and also open an issue to remove that entry once the next release of `pack` is published.

File: .github/.syncignore
Comment (mhdawson) [reply .github/.syncignore:2]: @paketo-community/builders-maintainers issue and commit pushed so this should be ready to land - @paketo-community/builders-maintainers

**Review by ForestEckhardt — APPROVED** [review]

**Pull request merged**
